packages/p2p-v2/src/pages/orders/screens/OrderDetails/OrderDetails.tsx

```diff
@@ -1,29 +1,118 @@
+                    <FullPageMobileWrapper
+                        className='p2p-v2-order-details'
+                        onBack={onReturn}
+                        renderFooter={() => <OrderDetailsCardFooter />}
+                        renderHeader={() => (
+                            <Text as='div' className='w-full flex items-center justify-between' size='lg' weight='bold'>
+                                {headerText}
+                                <Button
+                                    className='h-full p-0'
+                                    color='white'
+                                    data-testid='dt_p2p_v2_order_details_chat_button'
+                                    onClick={() => setShowChat(true)}
+                                    variant='contained'
+                                >
+                                    <ChatIcon className='mt-2' />
+                                </Button>
+                            </Text>
+                        )}
+                    >
+                        {shouldShowLostFundsBanner && (
+                            <InlineMessage
+                                className='w-fit mx-[1.6rem] mt-[1.6rem]'
+                                iconPosition='top'
+                                variant='warning'
+                            >
+                                <Text size='xs'>{warningMessage}</Text>
+                            </InlineMessage>
+                        )}
+                        <OrderDetailsCard />
+                    </FullPageMobileWrapper>
+                )}
+            </OrderDetailsProvider>
+        );
+    }
+
     return (
-        <div className='grid grid-cols-none lg:grid-cols-2'>
-            <OrdersChatSection
-                id={orderId}
-                isInactive={!!orderDetails?.isInactiveOrder}
-                otherUserDetails={orderDetails?.otherUserDetails}
-            />
-        </div>
+        <OrderDetailsProvider value={{ isErrorOrderInfo, orderDetails }}>
+            <div className='w-full'>
+                <PageReturn onClick={onReturn} pageTitle={headerText} weight='bold' />
+                <div className='p2p-v2-order-details'>
+                    {shouldShowLostFundsBanner && (
+                        <InlineMessage className='w-fit mb-6' variant='warning'>
+                            <Text size='2xs'>{warningMessage}</Text>
+                        </InlineMessage>
+                    )}
+                    <div className='grid grid-cols-none lg:grid-cols-2 lg:gap-14'>
+                        <OrderDetailsCard />
+                        <OrdersChatSection
+                            id={orderId}
+                            isInactive={!!orderDetails?.isInactiveOrder}
+                            otherUserDetails={orderDetails?.otherUserDetails}
+                        />
+                    </div>
+                </div>
+            </div>
+        </OrderDetailsProvider>
     );
 };
```

Comment:
@ameerul-deriv, can't we use one `<OrderDetailsProvider />` as well? 🤔

Reply:
hmmm i can check to see if there is some sort of work around

Reply:
the issue is because the UI in responsive looks different on desktop and the same with the behaviour

Reply:
ok i tried to separate it but on responsive theres a lot of issues when it comes to trying to render the orders chat, cause I have to add more logic to hide the Footer and Header if the chat is shown, which is already being handled in the FullPageMobileWrapper component above. And it adds even more logic to determine what to show on responsive and on desktop and depending whether or not I need to show the chat 😓 lets just leave it like this since its isolating the two into Mobile behaviour and Desktop behaviour